src/workflow/client/DaprWorkflowClient.ts

```diff
@@ -12,32 +12,51 @@ limitations under the License.
 */
 
 import { TaskHubGrpcClient } from "@microsoft/durabletask-js";
-import * as grpc from "@grpc/grpc-js";
 import { WorkflowState } from "./WorkflowState";
-import { generateInterceptors } from "../internal/ApiTokenClientInterceptor";
+import { generateApiTokenClientInterceptors } from "../internal/index";
 import { TWorkflow } from "../../types/workflow/Workflow.type";
 import { getFunctionName } from "../internal";
 import { Settings } from "../../utils/Settings.util";
+import { WorkflowClientOptions } from "../../types/workflow/WorkflowClientOption";
 
+/**
+ * Class that defines client operations for managing workflow instances.
+ *
+ * Instances of this class can be used to start, query, raise events to, and terminate workflow instances. In most
+ * cases, methods on this class accept an instance ID as a parameter, which identifies the workflow instance.
+ */
 export default class DaprWorkflowClient {
   private readonly _innerClient: TaskHubGrpcClient;
 
   /**
    * Initializes a new instance of the DaprWorkflowClient.
-   * @param {string | undefined} hostAddress - The address of the Dapr runtime hosting the workflow services.
-   * @param {grpc.ChannelOptions | undefined} options - Additional options for configuring the gRPC channel.
+   * @param {WorkflowClientOptions | undefined} options - Additional options for configuring DaprWorkflowClient.
    */
-  constructor(hostAddress?: string, options?: grpc.ChannelOptions) {
+  constructor(options: Partial<WorkflowClientOptions> = {}) {
+    const hostAddress = this.generateHostAddress(options);
+    options.daprApiToken = this.generateDaprApiToken(options);
     this._innerClient = this.buildInnerClient(hostAddress, options);
   }
 
-  private buildInnerClient(hostAddress?: string, options: grpc.ChannelOptions = {}): TaskHubGrpcClient {
-    const innerOptions = {
-      ...options,
-      interceptors: [generateInterceptors(), ...(options?.interceptors ?? [])],
-    };
-    if (hostAddress === undefined) {
-      hostAddress = Settings.getDefaultHost() + ":" + Settings.getDefaultGrpcPort();
+  private generateHostAddress(options: Partial<WorkflowClientOptions>): string {
+    const host = options?.clientHost ?? Settings.getDefaultHost();
+    const port = options?.clientPort ?? Settings.getDefaultGrpcPort();
+    const hostAddress = `${host}:${port}`;
+    return hostAddress;
+  }
+
+  private generateDaprApiToken(options: Partial<WorkflowClientOptions>): string | undefined {
```

Comment:
You can remove this one-liner function, or...
```suggestion
  private getDaprApiToken(options: Partial<WorkflowClientOptions>): string | undefined {
```